atlas/AtlasMl/atlasml/ml/pipeline_workflows.py

```
@@ -340,18 +256,138 @@
-            self.initial_cluster_pipeline()
-            self.initial_cluster_to_competency_pipeline()
+            self.recluster_with_new_competencies(competency=competency, course_id=competency.course_id)
         else:
             logger.warning(f"Competency with id {competency.id} not found for deletion")
 
+    def recluster_with_new_competencies(self, competency: Competency, course_id: str):
+        # Get competencies and exercise embeddings
+        competencies = self.weaviate_client.get_embeddings_by_property(
+            CollectionNames.COMPETENCY.value, "course_id", course_id
+        )
+        if len(competencies) == 0: return
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)